targets/targets.json

```diff
@@ -3221,9 +3222,13 @@
         "macros_add": [
             "STM32H743xx",
             "EXTRA_IDLE_STACK_REQUIRED",
-            "MBED_TICKLESS"
+            "MBED_TICKLESS",
+            "ETH_IP_VERSION_V2"
```

Comment:
cant find this used anywhere in this PR?

Reply:
This is used in features/netsocket/emac-drivers/TARGET_STM_EMAC/stm32xx_emac.cpp file. Used to differentiate H7 specific parts.

Reply:
Because github doesn't expand all files by default :-)

Reply:
I've missed that file ! I can see now. Wouldn't the config be better for this or ?

Reply:
You prefer to add this compilation switch in an other file ? Which one do you have in mind ? (features\netsocket\emac-drivers\TARGET_STM_EMAC\mbed_lib.json for example ?)

I need this switch to be activated for only H7 platform for now. Then, if this new hal is deployed on other platform, we will need to activate it also for those new platforms.

Reply:
This location seems correct to me - the target descriptor passing information to its drivers about which mode to select.

Reply:
you may move the stm32xx_emac_config.h file to the family specific folder and add this option there. this way you could tune the IP version per family (and later the thread stack size if needed)

Reply:
Code updated with @LMESTM solution. Effectively, user is not supposed to change this setting so the .json file can be misunderstanding.
Thanks all for comments and propositions !